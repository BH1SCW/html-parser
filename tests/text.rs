use html_parser::{HtmlParser, Result};
use indoc::indoc;
use insta::assert_debug_snapshot;

#[test]
fn it_can_parse_document_with_just_text() -> Result<()> {
    let markup = "hello world";
    let ast = HtmlParser::parse(markup)?;
    assert_debug_snapshot!(ast);
    Ok(())
}

#[test]
fn it_can_parse_document_with_text_and_line_breaks() -> Result<()> {
    let markup = indoc!(
        r"
        hello world
        here's another line for you!
        The end
    "
    );
    let ast = HtmlParser::parse(markup)?;
    assert_debug_snapshot!(ast);
    Ok(())
}

#[test]
fn it_can_parse_document_with_multiple_text_elements() -> Result<()> {
    let markup = indoc!(
        r"
        hello world
        here's another line for you!
        <div/>
        The end
    "
    );
    let ast = HtmlParser::parse(markup)?;
    assert_debug_snapshot!(ast);
    Ok(())
}
