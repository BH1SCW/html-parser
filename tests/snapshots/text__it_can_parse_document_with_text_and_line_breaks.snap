---
source: tests/text.rs
expression: ast
---
{
  "treeType": "documentFragment",
  "nodes": [
    {
      "text": "hello world\nhere's another line for you!\nThe end"
    }
  ]
}
